src/backend/distributed/commands/utility_hook.c

```diff
@@ -88,6 +96,8 @@ static void IncrementUtilityHookCountersIfNecessary(Node *parsetree);
 static void PostStandardProcessUtility(Node *parsetree);
 static void DecrementUtilityHookCountersIfNecessary(Node *parsetree);
 static bool IsDropSchemaOrDB(Node *parsetree);
+static void UndistributeDisconnectedCitusLocalTables(void);
```

Comment:
1) implement a udf for that static function
2) error out for create_citus_local_table when `ShouldEnableForeignKeys` returns false

Reply:
2. -> 0c86fc11a40e8845dc37d95f2343871f15d20d60